azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+	if err != nil {
+		return fmt.Errorf("Error retrieving MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read MariaDB Server %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmMariaDbServerRead(d, meta)
+}
+
+func resourceArmMariaDbServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[WARN] MariaDB Server %q was not found (resource group %q)", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Azure MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	d.Set("administrator_login", resp.AdministratorLogin)
+	d.Set("version", string(resp.Version))
+	d.Set("ssl_enforcement", string(resp.SslEnforcement))
+
+	if err := d.Set("sku", flattenMariaDbServerSku(resp.Sku)); err != nil {
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+
+	if err := d.Set("storage_profile", flattenMariaDbStorageProfile(resp.StorageProfile)); err != nil {
+		return fmt.Errorf("Error setting `storage_profile`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	// Computed
+	d.Set("fqdn", resp.FullyQualifiedDomainName)
+
+	return nil
+}
+
+func resourceArmMariaDbServerDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).mariadbServersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["servers"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+
+		return fmt.Errorf("Error deleting MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+
+		return fmt.Errorf("Error waiting for deletion of MariaDB Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandAzureRmMariaDbServerSku(d *schema.ResourceData) *mariadb.Sku {
+	skus := d.Get("sku").([]interface{})
+	sku := skus[0].(map[string]interface{})
+
+	name := sku["name"].(string)
+	capacity := sku["capacity"].(int)
+	tier := sku["tier"].(string)
+	family := "Gen5"
```

Comment:
IMO we should expose this and make it a Required field, even if there's only one value for the moment (since in future when other generations exist this will be an odd default, imo?)

Reply:
Added family back into scheam and read it from the struct on expand.